### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by billmn**
Added Twig configurations and driver improvements :
- Added all configurations supported by Twig Environment
- Added configurations for Twig Lexer
- Changed Twig_Loader_String to Twig_Loader_Filesystem for template inheritance

### CODE REVIEW

**Comment by jschreuder** [comment]
Why does the Twig dir have to be uppercase first char? You create the dir & copy in those files yourself don't you? I don't see a good reason to break our conventions before we're actually in the lib.

Also I'd prefer it if you kept the static properties/methods on top and the dynamic (non-static) below, but I can change that back myself.

On the whole: great job, thanks for contributing!

**Comment by billmn** [comment]
With lowercase "lib_name" ( also with your twig driver ) I obtain :

"Class 'Twig_Loader_String' not found"

I think there are some problems with Twig interfaces ... if you use "Twig" all works fine.
In Twig's documentation ( http://www.twig-project.org/doc/intro.html ) the lib name always appear with first char uppercased.

I can move dynamic properties below.
I had to move Twig_Load_Filesystem on "capture" function because needs view name

File: classes/view/twig.php [hunk 1]
```diff
@@ -16,47 +16,63 @@ namespace Parser;
...
+        \Twig_Autoloader::register();
+    }
+
+    protected static function capture($view_filename, array $view_data)
+    {
+        $data = static::$_global_data;
+        $data = array_merge($data, $view_data);
+        
+        // Extract View name/extension (e.g. "template.twig")
+        $view_name = pathinfo($view_filename, PATHINFO_FILENAME) . '.' . pathinfo($view_filename, PATHINFO_EXTENSION);
```
Comment (jschreuder) [inline classes/view/twig.php]: Why not just use PATHINFO_BASENAME?

File: classes/view/twig.php [hunk 2]
```diff
@@ -16,47 +16,63 @@ namespace Parser;
...
+        
+        // Twig Loader
+        $views_paths = \Config::get('parser.View_Twig.views_paths', array(APPPATH . 'views'));
+        \Arr::insert($views_paths, pathinfo($view_filename, PATHINFO_DIRNAME), 0);
```
Comment (jschreuder) [inline classes/view/twig.php]: You don't need to use `\Arr::insert()` here and you can save the dependency by using `array_unshift($view_paths, pathinfo($view_filename, PATHINFO_DIRNAME))` instead.

File: classes/view/twig.php [hunk 3]
```diff
@@ -16,47 +16,63 @@ namespace Parser;
...
+        {
+			// @TODO: Some problems with Twig/Fuel exceptions
+            ob_end_clean();	// Delete the output buffer ( silently )
+            throw $e;		// Re-throw the exception
+        }
+    }
+    
+    public function parser()
+    {
+        if (!empty(static::$_parser))
```
Comment (jschreuder) [inline classes/view/twig.php]: The ! requires spaces on both sides from our coding guidelines. Thus `if ( ! empty(static::$_parser))`

**Comment by jschreuder** [comment]
You're right about the Twig autoloader, it demands its basedir to be named "Twig", it moves one dir up when the classname starts with "Twig_" and then includes "Twig/" into the expected dir - thus disallowing renaming the dir it's in.

Other than my notes in the code it's ready for merging.

File: classes/view/twig.php (hunk 1 above)
Comment (billmn) [inline classes/view/twig.php]: You're right ... stupid code

**Comment by billmn** [comment]
I have commited changes ... take a look

**Comment by jschreuder** [comment]
Great, thanks!

**Pull request merged**